std/range/package.d

```diff
@@ -8146,433 +8306,824 @@ public:
+        assert(iota(1, 5).slide!Partial(4)[0 .. 1].equal!equal([[1, 2, 3, 4]]));
+    }
+
+    // special cases
+    assert(iota(3).slide!(Yes.withPartial)(4)[0 .. $].equal!equal([[0, 1, 2]]));
+    assert(iota(3).slide!(No.withPartial)(4)[0 .. $].empty);
+}
 
-    assert(iota(3).slide!(No.withFewerElements)(4).empty);
-    assert(iota(3, 3).slide!(No.withFewerElements)(4).empty);
-    assert(iota(3).slide!(No.withFewerElements)(4).length == 0);
-    assert(iota(3).slide!(No.withFewerElements)(4, 4).length == 0);
+// test slicing (Yes.withPartial)
+@safe pure nothrow unittest
+{
+    import std.algorithm.comparison : equal;
 
-    assert(iota(3).slide!(No.withFewerElements)(400).empty);
-    assert(iota(3).slide!(No.withFewerElements)(400).length == 0);
-    assert(iota(3).slide!(No.withFewerElements)(400, 10).length == 0);
+    auto r = 7.iota.slide!(Yes.withPartial)(2, 3);
 
-    assert(iota(3).slide!(No.withFewerElements)(4)[0 .. $].empty);
-    assert(iota(3).slide!(No.withFewerElements)(4)[$ .. $].empty);
-    assert(iota(3).slide!(No.withFewerElements)(4)[$ .. 0].empty);
-    assert(iota(3).slide!(No.withFewerElements)(4)[$/2 .. $].empty);
+    assert(r[0 .. 1].equal!equal([[0, 1]]));
+    assert(r[0 .. 2].equal!equal([[0, 1], [3, 4]]));
+    assert(r[0 .. 3].equal!equal([[0, 1], [3, 4], [6]]));
+
+    assert(r[1 .. 1].empty);
+    assert(r[1 .. 2].equal!equal([[3, 4]]));
+    assert(r[1 .. 3].equal!equal([[3, 4], [6]]));
+
+    assert(r[2 .. 2].empty);
+    assert(r[2 .. 3].equal!equal([[6]]));
+    assert(r[3 .. 3].empty);
+}
+
+// test slicing (No.withPartial)
+@safe pure nothrow unittest
+{
+    import std.algorithm.comparison : equal;
+
+    auto r = 7.iota.slide!(No.withPartial)(2, 3);
+
+    assert(r[0 .. 1].equal!equal([[0, 1]]));
+    assert(r[0 .. 2].equal!equal([[0, 1], [3, 4]]));
+
+    assert(r[1 .. 1].empty);
+    assert(r[1 .. 2].equal!equal([[3, 4]]));
+
+    assert(r[2 .. 2].empty);
+}
+
+// length
+@safe pure nothrow unittest
+{
+    import std.algorithm.comparison : equal;
+
+    static foreach (Partial; [Yes.withPartial, No.withPartial])
+    {
+        assert(iota(3).slide!Partial(1).length == 3);
+        assert(iota(3).slide!Partial(1, 2).length == 2);
+        assert(iota(3).slide!Partial(1, 3).length == 1);
+        assert(iota(3).slide!Partial(1, 4).length == 1);
+        assert(iota(3).slide!Partial(2).length == 2);
+        assert(iota(3).slide!Partial(2, 3).length == 1);
+        assert(iota(3).slide!Partial(3).length == 1);
+        assert(iota(3).slide!Partial(3, 2).length == 1);
+    }
+
+    // length special cases
+    assert(iota(3).slide!(Yes.withPartial)(2, 2).length == 2);
+    assert(iota(3).slide!(No.withPartial)(2, 2).length == 1);
+}
+
+// opDollar
+@safe pure nothrow unittest
+{
+    import std.algorithm.comparison : equal;
+
+    static foreach (Partial; [Yes.withPartial, No.withPartial])
+    {
+        assert(iota(4).slide!Partial(4)[$/2 .. $].equal!equal([[0, 1, 2, 3]]));
```

Comment:
You test the behavior of opDollar in other places, so this unittest block ends up being redundant